crates/host/src/wasmbus/mod.rs

```diff
@@ -1361,16 +1360,31 @@ impl Host {
         self.store_component_spec(&component_id, &component_spec)
             .await?;
 
+        let links = component_import_links(&component_spec.links);
+        // TODO: We need to support multiple link names with this. Right now this only supports default
+        let maybe_messaging = if let Some(link) = links.get("default").and_then(|links| links.get("wasmcloud:messaging/consumer")) {
+            // Hack: This is how we know if it is a host ID rather than another NKEY. We should
+            // replace this and not require the host ID for every component
+            if link.starts_with("N") {
```

Comment:
I thought from one of the tests that we were using the Host ID, but it is actually still just ID of the provider. This code needs to be updated to go fetch the details about the provider from KV and check if it is a builtin (or do it another way)